### PR TITLE
Fix debug_defines make process.

### PR DESCRIPTION
**Pull request by timsifive**
With the old version I ended up with a debug_defines.h that had many copies of itself. Have registers.py always generate the whole thing, so there is no chance of repeated executions simply appending again and again.

### CODE REVIEW

**Comment by timsifive** [comment]
@en-sc, can you review this? I'll get Paul to do the official sign-off once you approve.

**Comment by en-sc** [comment]
LGTM (the only comment is more of a nitpick).

**Comment by timsifive** [comment]
> LGTM (the only comment is more of a nitpick).

I don't see any comments.

**Review by en-sc — COMMENTED** [review]
Sorry, I forgot to submit it yesterday.

File: Makefile [hunk 1]
```diff
@@ -62,15 +62,13 @@ changelog.tex: .git/logs/HEAD Makefile
...
-	echo " * This file is auto-generated by running 'make debug_defines' in" >> $@
-	echo " * https://github.com/riscv/riscv-debug-spec/ (`git describe --always --dirty --exclude '*'`)" >> $@
-	echo " */" >> $@
-	echo >> $@
+.PHONY: debug_defines
+debug_defines: debug_defines.h debug_defines.c
+
+debug_defines.c: debug_defines.h
+
+debug_defines.h: $(patsubst %,xml/%,$(REGISTERS_TEX:.tex=.xml)) registers.py
```
[the remark this answers is not in the document]
Comment (timsifive) [reply Makefile:70]: The goal is to have the following work:
* `make debug_defines.c` make the .c file
* `make debug_defines.h` make the .h file

That's just how I feel Makefiles ought to work. They have targets for specific files.

Then `make debug_defines` is a convenience target. Doing it this way also means that registers.py is only run when debug_defines.[ch] are out of date.

[the remark this answers is not in the document]
Comment (en-sc) [reply Makefile:70]: Oh, I see the point. Maybe you would prefer this then (a bit more symmetrical):
```suggestion
.PHONY: debug_defines
debug_defines: debug_defines.h debug_defines.c

debug_defines.h debug_defines.c: $(patsubst %,xml/%,$(REGISTERS_TEX:.tex=.xml)) registers.py
```

[the remark this answers is not in the document]
Comment (timsifive) [reply Makefile:70]: I had already tried that. In that case `make` will run registers.py twice in parallel (if you have parallel make enabled) when neither file exists.

[the remark this answers is not in the document]
Comment (en-sc) [reply Makefile:70]: Thanks for the explanation!

**Comment by en-sc** [comment]
LGTM

**Comment by rtwfroody** [comment]
I'm abandoning this change. The main bug was already fixed in the asciidoc migration. The .c and .h targets aren't that useful individually, even if it does some "more right".

**Pull request closed without merging**